service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsControllerTest.java

```diff
@@ -0,0 +1,125 @@
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(HearingBookingDetailsController.class)
+@OverrideAutoConfiguration(enabled = true)
+class HearingBookingDetailsControllerTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    private static final String ERROR_MESSAGE = "Enter a future date";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    void shouldReturnAnErrorWhenHearingDateIsSetToYesterday() throws Exception {
+        LocalDate yesterday = LocalDate.now().minusDays(1);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(createHearing(yesterday));
+
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldNotReturnAnErrorWhenHearingDateIsSetToTomorrow() throws Exception {
+        LocalDate tomorrow = LocalDate.now().plusDays(1);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(createHearing(tomorrow));
+
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenHearingDateIsSetToToday() throws Exception {
+        LocalDate today = LocalDate.now();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(createHearing(today));
+
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenHearingDateIsSetInDistantPast() throws Exception {
+        LocalDate distantPast = LocalDate.now().minusYears(10000);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(createHearing(distantPast));
+
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldNotReturnAnErrorWhenHearingDateIsSetInDistantFuture() throws Exception {
+        LocalDate distantFuture = LocalDate.now().plusYears(1000);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(createHearing(distantFuture));
+
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+    }
+
+    private HearingBooking createHearing(LocalDate hearingDate) {
+        return HearingBooking.builder()
+            .hearingDate(hearingDate)
+            .build();
+    }
+
+    @SuppressWarnings("LineLength")
+    private AboutToStartOrSubmitCallbackResponse makeRequest(HearingBooking hearingDetail) throws Exception {
```

Comment:
What about creating the helper method to run this code? Of course it would require a bit of refactoring, but I have noticed we are doing similar methods in our various tests, don't need to happen right now, but worth noting for the future :).

Reply:
👍